### ISSUE TITLE
E722 BaseException should also be considered as bare except

### ISSUE THREAD

**Issue by typenoob** [astral-sh/ruff#21518]
### Summary

According to the document,
> A bare except catches BaseException which includes KeyboardInterrupt, SystemExit, Exception, and others. Catching > BaseException can make it hard to interrupt the program (e.g., with Ctrl-C) and can disguise other problems.

A bare except is equal with catching BaseException explicitly.
Therefore we should report the following snippets as well.
```python
try:
    pass
except BaseException:
    pass
```
Does it make sense?

**Comment by amyreese** [comment]
Seems like a reasonable thing to catch, IMO.

**Comment by ntBre** [comment]
I think [blind-except (BLE001)](https://docs.astral.sh/ruff/rules/blind-except/) is meant to catch this case.  We should probably cross-link between the two rules at least.

**Comment by typenoob** [comment]
BLE001 also catches the ```except Exception:``` which conflicts with the E722's suggestion.
> If you actually need to catch an unknown error, use Exception which will catch regular program errors but not important system
> exceptions.

> ```py
> def run_a_function(some_other_fn):
>    try:
>        some_other_fn()
>    except Exception as e:
>        print(f"How exceptional! {e}")
> ```

**Comment by Avasam** [comment]
Depends on the intention for the rule: is a bare `except` bad because `BaseException` is bad? Or is it because most devs not knowledgeable about Python's exception internals won't even know that `except` isn't `except Exception` ? And avoiding bare excepts helps clarify intent whilst avoiding accidentally eating up Keyboard interrupts.

If you write `except BaseException`, you are being explicit about your intentions.

> which conflicts with the E722's suggestion.

I don't see them conflicting. One helps you clarify intent, the other then tells you you may be hiding away unexpected errors (rather than logging or rethrowing)

**Comment by typenoob** [comment]
> I don't see them conflicting. 

I mean they can't use together. E722 allow **Exception**, but BLE001 don't. So I can't combine them to a point where **BaseException** and **bare expect** is not allowed but **Exception** is allowed. 

And since E772's intention is described as below

>A bare except catches BaseException which includes KeyboardInterrupt, SystemExit, Exception, and others. Catching BaseException can make it hard to interrupt the program (e.g., with Ctrl-C) and can disguise other problems.

IMO, in E772, **BaseException** is eqaul to **bare except** and should be treat equally. If you think they have slight difference, then it's fine to stay still. Though I can't get it through reading the docs.

**Comment by typenoob** [comment]
>  is a bare except bad because BaseException is bad? Or is it because most devs not knowledgeable about Python's exception internals won't even know that except isn't except (Exception) ?

So what's the intention actually? Since its name is **bare-except**, I prefer the latter. But without your explanation, I would have stuck with the former.
